sklearn/neighbors/base.py

```diff
@@ -417,6 +407,26 @@ class from an array representing our data set and ask who's
                 return dist, neigh_ind
             return neigh_ind
 
+    def _reduce_func(self, dist, n_samples, **kwds):
```

Comment:
you should be able to accept parameters by name here, i.e. `n_neighbors`, `return_distance` should be listed explicitly.